diff --git a/src/utility/rtc/RTC_PowerHub_Class.cpp b/src/utility/rtc/RTC_PowerHub_Class.cpp
index 37ec1ab..821e4f8 100644
--- a/src/utility/rtc/RTC_PowerHub_Class.cpp
+++ b/src/utility/rtc/RTC_PowerHub_Class.cpp
@@ -92,7 +92,7 @@ namespace m5
   int RTC_PowerHub_Class::setAlarmIRQ(const rtc_date_t *date, const rtc_time_t *time)
   {
 
-    std::uint8_t buf[3];
+    std::uint8_t buf[3] = {0, 0, 0};
     bool irq_enable = false;
 
     if (time) {
